Index: bcc-2.0.4/ubuild.sh
===================================================================
--- bcc-2.0.4.orig/ubuild.sh
+++ bcc-2.0.4/ubuild.sh
@@ -36,7 +36,7 @@ build_log () {
   echo "> opt=$opt" >> $logdir/$fn.stdout
   echo "> $@" >> $logdir/$fn.stdout
   `bash -c "$*" 1>> $logdir/$fn.stdout 2>> $logdir/$fn.stderr` || \
-    if [ $?!= 0 ]; then
+    if [ "$?" != 0 ]; then
       echo "FAIL: See $logdir/$fn.stderr"
       exit 1
     fi
@@ -69,6 +69,7 @@ build_toolchain () {
       --disable-nls \
       --disable-gdb \
       --disable-sim \
+      --disable-werror \
       --with-sysroot=$opt/$TARGET \
       --with-build-sysroot=$bdir/$opt/$TARGET \
       --enable-plugins"
Index: bcc-2.0.4/binutils/opcodes/sparc-opc.c
===================================================================
--- bcc-2.0.4.orig/binutils/opcodes/sparc-opc.c
+++ bcc-2.0.4/binutils/opcodes/sparc-opc.c
@@ -1038,6 +1038,19 @@ const struct sparc_opcode sparc_opcodes[
 { "subccc",	F3(2, 0x1c, 0), F3(~2, ~0x1c, ~0)|ASI(~0),	"1,2,d", 0, 0, 0, v9 },
 { "subccc",	F3(2, 0x1c, 1), F3(~2, ~0x1c, ~1),		"1,i,d", 0, 0, 0, v9 },
 
+{ "andc8",	F3(3, 0x08, 0), F3(~3, ~0x08, ~0)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
+{ "andc16",	F3(3, 0x0b, 0), F3(~3, ~0x0b, ~0)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
+{ "xorc8",	F3(3, 0x0c, 0), F3(~3, ~0x0c, ~0)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
+{ "xorc16",	F3(3, 0x0e, 0), F3(~3, ~0x0e, ~0)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
+{ "xnorc8",	F3(3, 0x18, 0), F3(~3, ~0x18, ~0)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
+{ "xnorc16",	F3(3, 0x1b, 0), F3(~3, ~0x1b, ~0)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
+
+{ "tibs",	F3(2, 0x1d, 0), F3(~2, ~0x1d, ~1)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
+{ "invtibs",	F3(2, 0x19, 0), F3(~2, ~0x19, ~1)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
+{ "red",	F3(2, 0x0d, 1), F3(~2, ~0x0d, ~1),		"1,i,d", 0, 0, 0, v6 },
+{ "ftchk",	F3(2, 0x2e, 1), F3(~2, ~0x2e, ~1),		"1,i,d", 0, 0, 0, v6 },
+{ "tibsrot",	F3(2, 0x09, 1), F3(~2, ~0x09, ~1),		"1,i,d", 0, 0, 0, v6 },
+
 { "and",	F3(2, 0x01, 0), F3(~2, ~0x01, ~0)|ASI(~0),	"1,2,d", 0, 0, 0, v6 },
 { "and",	F3(2, 0x01, 1), F3(~2, ~0x01, ~1),		"1,i,d", 0, 0, 0, v6 },
 { "and",	F3(2, 0x01, 1), F3(~2, ~0x01, ~1),		"i,1,d", 0, 0, 0, v6 },
